mautrix/util/ffmpeg.py

```diff
@@ -0,0 +1,141 @@
+# Copyright (c) 2021 Tulir Asokan
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+from __future__ import annotations
+
+from typing import Iterable
+from pathlib import Path
+import asyncio
+import mimetypes
+import os
+import shutil
+import tempfile
+
+try:
+    from . import magic
+except ImportError:
+    magic = None
+
+
+def _abswhich(program: str | None) -> str | None:
+    path = shutil.which(program)
+    return os.path.abspath(path) if path else None
+
+
+class ConverterError(ChildProcessError):
+    pass
+
+
+class NotInstalledError(ConverterError):
+    def __init__(self) -> None:
+        super().__init__("ffmpeg not found in PATH")
+
+
+ffmpeg_path = _abswhich("ffmpeg")
+ffmpeg_default_params = ("-hide_banner", "-loglevel", "warning")
+
+
+async def convert_path(
+    input_file: Path | str,
```

Comment:
maybe use https://docs.python.org/3.8/library/os.html#os.PathLike?

```suggestion
    input_file: PathLike,
```